### ISSUE TITLE
Node ports starting with a "compass letter" breaks parsing

### ISSUE THREAD

**Issue by acidghost** [besok/graphviz-rust#24]
The following graph is not parsed correctly:
```dot
digraph "test" {
    A:s0 -> B;
}
```
It's parsed as
```
DiGraph {
    id: Escaped("\"test\""), strict: false,
    stmts: [
        Node(Node { id: NodeId(Plain("A"), Some(Port(None, Some("s")))), attributes: [] }),
        Edge(Edge { ty: Pair(N(NodeId(Plain("0"), None)), N(NodeId(Plain("B"), None))), attributes: [] })
    ]
 }
```

while I would expect

```
DiGraph {
    id: Escaped("\"test\""), strict: false,
    stmts: [
        Edge(Edge { ty: Pair(N(NodeId(Plain("A"), Some(Port(Some(Plain("s0")), None)))), N(NodeId(Plain("B"), None))), attributes: [] })
    ]
}
```

I'm able to fix this by swapping the parsing expressions for `port`
https://github.com/besok/graphviz-rust/blob/b9593c6f0f0666c0a229a03d204ddc07f6e10efc/src/grammar/dot.pest#L19
to
```
port = { (":" ~ id ~ (":" ~ compass)?) | (":" ~ compass) }
```
but this seems to fail to parse actual "compass" nodes (e.g. `nodeId:n`, `nodeId:s`, etc).

**Comment by besok** [comment]
Yeah, it seems, there is an ambiguity in the grammar. 
One way is to rewrite the grammar with an attempt to avoid the ambiguity or another way is to try to resolve it after in a code.
I can take a look later this week.

**Comment by besok** [comment]
should be fixed in `0.7.0` version https://github.com/besok/graphviz-rust/commit/d9e1ea03ae432e90ce670801fd71c7b31102a771
